app/vmalert/rule/group.go

```diff
@@ -310,17 +285,43 @@ func (g *Group) Close() {
 	g.InterruptEval()
 	<-g.finishedCh
 
-	g.metrics.close()
+	g.closeGroupMetrics()
 }
 
 // SkipRandSleepOnGroupStart will skip random sleep delay in group first evaluation
 var SkipRandSleepOnGroupStart bool
 
+// startGroupMetrics creates metrics for group and its rules, and register the set
+func (g *Group) startGroupMetrics() {
```

Comment:
My main problem with this function, that it looks optional. But in fact, it's required to use `Group` methods, otherwise it may panic. And `Init` looks more natural to me.

 Also, calling `Init` method outside `Group.Start` method makes it's easier to use. Because additional synchronization is required after `Group.Start`, if it is called inside goroutine.

Reply:
Updated, please check https://github.com/VictoriaMetrics/VictoriaMetrics/pull/8534/commits/d948043af07195e89d898e7513b17ae6aeace2ae.